### DISCUSSION TITLE
pockethost.io paid tier discussion

### DISCUSSION THREAD

**Discussion by benallfree** [pockethost/pockethost#287] [Q&A]
Hey everyone, I wanted to open a discussion about a paid tier for pockethost.io and what everyone felt would be valuable and equitable. 

First, a reminder - the PocketHost project is MIT open source. I think it is the only open source multitenant pocketbase hosting solution. Anyone can take this project and run it for private or public use. It's badass.

This discussion is about [pockethost.io](https://pockethost.io), which I've been running at my own expense for about a year. As you can see from the [system health megathread](#223), outages have been rare and fixed quickly. We are tracking a few outage causes that seem to show up only in production. For example, the latest one is a Knex timeout error when accessing the raw SQLite files. The only option is to wait for these to surface and fix them as they come. Dogfooding the PocketHost project on pockethost.io has been instrumental in creating a stable platform.

Here some pockethost.io stats:

| name                     |  value                 |
|------------------------|---------------------|
| userCount              | 2636                |
| runningInstanceCount   | 14                  |
| instanceCount          | 3560                |
| instanceCount1Hour     | 82                  |
| instanceCount1Day      | 297                 |
| instanceCount7Day      | 617                 |
| instanceCount30Day     | 1159                |
| invocationCount        | 2613110             |
| invocationCount1Hour   | 606                 |
| invocationCount1Day    | 15068               |
| invocationCount7Day    | 133707              |
| invocationCount30Day   | 447101              |
| invocationSeconds      | 82116629.66801019   |
| invocationSeconds1Hour | 64166.25600000002   |
| invocationSeconds1Day  | 842562.3680000034   |
| invocationSeconds7Day  | 5469632.9049999295  |
| invocationSeconds30Day | 18793014.398999963  |

Note: an _instance_ is a specific subdomain. An _invocation_ is a launching of the `pocketbase` executable.

Please check my math, but here's what I see:
* ~2,600 users have created ~3,500 instances
* In any given hour, ~80 unique instances are accessed (http requests)
* The `pocketbase` executable (any version) has been launched 2,613,110 times, and about 600 times per hour
* In the last 30 days, ~1,100 unique instances have been accessed. That's about 30% of all instances. That's pretty remarkable.
* In the last 30 days, the combined CPU run time of all `pocketbase` executables was 18.7 million seconds. That's 313k hours.

These stats are extra impressive to me because PocketHost automatically pauses instances after 5s of idle time. So it's not like instances are running and not being used. These stats are based on real usage. The service is bringing value to a lot of people. It fills my heart with the good feelz 🐱  

Other fun facts:
* pockethost.io is running on a Digital Ocean droplet: 8 vCPUs, 16 GB RAM, 320 GB storage, 6 TB bandwidth.
* CPU usage is rarely above 20%, which means that every `pocketbase` instance effectively has the machine to itself. I'll keep expanding resources so it stays that way.
* We are running in just one region - San Francisco USA. On the Discord server, there have been some people discussing unacceptable latency from across the globe, but when we try to dig deeper into the causes of latency, the users have stopped responding and the trail runs cold. It's not yet clear to me that offering regional support for PocketHost is a benefit to anyone. If we can show clearly that running PocketHost in several regions actually does reduce latency in a meaningful way, we can add it.

I ran a quick query and **there are about 200 power users whose instances are running for >100 minutes per month and about 2,000 casual users running <=100 minutes per month**

If I implemented the long-discussed free tier usage rule of 100 minutes per month, it looks like about 10% of the users would hit the usage limit. I have no data on how many of them would move to a paid tier, but here's my first thought:

* Free Tier - 100 minutes per month
* Paid Tier - $20/mo for unlimited minutes and instances

That seems fine to me, rather than granular tracking of resources. We don't have any abusers or bad neighbors on the system, so why solve a problem we don't have? A paid tier looks like it might bring in $1-3k of monthly revenue, which would allow me to keep hosting resources provisioned well above usage. And if someone ever did really begin consuming a lot, I could move them to a separate "Enterprise" plan or dedicated resources.

Let me know your thoughts, guys. I'm curious to know how you see the service evolving. We've gotten lots of questions about service level agreements and production stability, maybe it's time to address it.

**Comment by StepanKukharskiy** [comment]
Hey there! 
Seems like I will be among 10% switching to a paid tier. My instance usage is way over 100 minutes with just 30+ users active in the moment. Is usage number refreshed every month? 

![IMG_1431](https://github.com/benallfree/pockethost/assets/47927208/55b6a745-e260-44ff-a4ae-e00a4127de8f)

**Reply by benallfree** [reply]
Yes, that would be your usage this month. Does that seem accurate?

**Reply by StepanKukharskiy** [reply]
I think so. And I think I know why 😅

**Reply by BasToTheMax** [reply]
Almost 1k minutes with just 30 users. Wow!

**Comment by benallfree** [comment]
One thought - anyone using realtime connections will quickly eat up their monthly minutes because their instance will stay running as long as any frontend maintains an SSE connection.

**Comment by BasToTheMax** [comment]
What payment methods would be available?

**Reply by benallfree** [reply]
How about Stripe?

**Comment by BasToTheMax** [comment]
I think 100 minutes isn't enough. Yesterday I played a bit with PocketHost and i'm already at 25 minutes. With just one user! (me)

**Reply by benallfree** [reply]
Ah. I think that’s my fault, but it raises a good point. I made an adjustment the other day where it sleeps after 60s idle time instead of 5s. Under that old model, you’d probably show under 1 minute of usage.

**Comment by benallfree** [comment]
Some further discussion: as an incremental step, I think I might implement usage limits and see who wants to move to the paid tier. That will help me gauge whether it is important to write subscription code. 

So:

Phase 1: implement 100 minute usage limit - get feedback, manually approve users who exceed it, possibly moving them to a paid tier

Phase 2: if enough users want to move to a paid tier, then I'll implement some automation
